mergekit/scripts/extract_lora.py

```diff
@@ -0,0 +1,194 @@
+import json
+import os
+
+import bitsandbytes as bnb
+import click
+import torch
+from peft.tuners.lora import QuantLinear
+from safetensors.torch import save_file
+from tqdm import tqdm
+from transformers import AutoConfig, AutoModelForCausalLM
+
+from mergekit.common import ModelReference
+from mergekit.io import LazyTensorLoader
+from mergekit.options import add_merge_options
+
+
+def _low_rank_decomposition(weight, reduced_rank=16):
```

Comment:
A little nitpicky but it would be great to have type annotations on these helper functions.